diff --git a/lisp/emacs-lisp/comp.el b/lisp/emacs-lisp/comp.el
index cac63a5978..6b7616ccba 100644
--- a/lisp/emacs-lisp/comp.el
+++ b/lisp/emacs-lisp/comp.el
@@ -2597,7 +2597,7 @@ Return the its filename if found or nil otherwise."
 ;;;###autoload
 (defun comp-subr-safe-advice (subr-name)
   "Make SUBR-NAME effectively advice-able when called from native code."
-  (unless (or (memq subr-name comp-never-optimize-functions)
+  (unless (or t (memq subr-name comp-never-optimize-functions)
               (gethash subr-name comp-installed-trampolines-h))
     (let ((trampoline-sym (comp-trampoline-sym subr-name)))
       (cl-assert (subr-primitive-p (symbol-function subr-name)))
